php/exceptions/UploadException.class.inc

```diff
@@ -0,0 +1,80 @@
+<?php declare(strict_types=1);
+/**
+ * This class is a wrapper that takes an upload error code
+ * an adds the appropriate message to the exception.
+ *
+ * PHP Version 7
+ *
+ * @category Errors
+ * @package  Loris
+ * @author   Xavier Lecours Boucher <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://github.com/aces/Loris
+ */
+
+/**
+ * The UploadException class
+ *
+ * @category Errors
+ * @package  Loris
+ * @author   Xavier Lecours Boucher <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://github.com/aces/Loris
+ */
+class UploadException extends \LorisException
+{
+    /**
+     * Contructor that takes the upload error code to generate the message.
+     *
+     * @param int $code The error code
+     *
+     * @see http://php.net/manual/en/features.file-upload.errors.php
+     */
+    public function __construct(int $code)
+    {
+        $message = $this->_codeToMessage($code);
+        parent::__construct($message, $code);
+    }
+
+    /**
+     * Mapper function to retreive the corresponding error message.
+     *
+     * @param int $code The error code
+     *
+     * @return string
+     */
+    private function _codeToMessage(int $code) : string
+    {
+        switch ($code) {
+        case UPLOAD_ERR_INI_SIZE:
+            $message = 'The uploaded file exceeds the upload_max_filesize ' .
+                       'directive in php.ini';
+            break;
+        case UPLOAD_ERR_FORM_SIZE:
+            $message = 'The uploaded file exceeds the MAX_FILE_SIZE ' .
```

Comment:
@cmadjar says:

> not sure I completely understand this error. Where does it come from?

Reply:
I think you wrote this code so I'm not sure.

Reply:
Those are php constants described at https://www.php.net/manual/en/features.file-upload.errors.php . I will let you write the appropriate message if it needs to be clarified.

Reply:
No that makes sense to me. 

MAX_FILE_SIZE is something you can put in HTML to limit how a file is uploaded. 

It's completely non-standard though and specific to PHP; I don't think anyone uses it.